proposals/Fleetctl-Docker-Image.md

```diff
@@ -0,0 +1,51 @@
+# Goal
+
+We need `fleetctl package` functionality to generate all types of packages (PKG, MSI, DEB and RPM) from Linux.
+
+# How
+
+Create a new Docker image `fleetdm/fleetctl` that will contain `fleetctl` and all the dependencies ready to create packages.
+
+Users can then use the image to generate packages
+```sh
+$ docker run ... fleetdm/fleetctl:latest package --type={pkg|msi|deb|rpm} ...
```

Comment:
per our discussion the user should also provide a `--native` flag here, is that correct?

Reply:
Can we look for the dependencies on the local system and just use them if they exist?

Reply:
> per our discussion the user should also provide a --native flag here, is that correct?

I was thinking more on setting an environment variable `FLEECTL_NATIVE=1` in the `Dockerfile` instead.
(That way the user of the docker image doesn't have to remember using such flag.)

> Can we look for the dependencies on the local system and just use them if they exist?

Am worried about reproducibility if we make that behavior the default. If users have different versions or configurations in their environments then it will be hard for us to troubleshoot and help them.